tests/restapi/test_threaded_restapi.py

```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+# pylint: disable=protected-access
+"""Tests for the `aiida.restapi` module, using it in threaded mode.
+
+Threaded mode is the default (and only) way to run the AiiDA REST API (see `aiida.restapi.run_api:run_api()`).
+This test file's layout is inspired by https://gist.github.com/prschmid/4643738
+"""
+import time
+from threading import Thread
+
+import requests
+from werkzeug.serving import make_server
+import pytest
+
+from aiida.restapi.common.config import CLI_DEFAULTS, API_CONFIG
+from aiida.restapi.run_api import configure_api
+
+# The URL to the REST API we will start up.
+# This is the default location where Flask apps start when using the built in web server.
+SERVER_URL = 'http://{hostname}:{port}{api}'.format(
+    hostname=CLI_DEFAULTS['HOST_NAME'], port=CLI_DEFAULTS['PORT'], api=API_CONFIG['PREFIX']
+)
+
+NO_OF_REQUESTS = 1000
+
+
+@pytest.mark.usefixtures('clear_database_before_test')
+def test_run_threaded_server(aiida_localhost):
```

Comment:
Not sure if there is a particular reason why you opted for this form, but to "use" a fixture, you just add it as an argument to the test signature. I know ... pytest magic :eyes: 
So these two lines could simply become
```
def test_run_threaded_server(clear_database_before_test, aiida_localhost):
```

Reply:
Actually kinda like this, at least `pylint` stops complaining about unused function arguments 😄 

But yeah, I think `usefixtures` is meant mostly for defining a fixture use for an entire module or class.

Reply:
It was to avoid adding a `pylint` disable as well ...